probe_src/performance_test.py

```diff
@@ -1,200 +1,228 @@
+        print("Removing log file.")
+        subprocess.run("rm probe_log", shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    if os.path.exists(RECORD_DIR):
+        print("Removing record directory.")
+        shutil.rmtree(RECORD_DIR)
+    time.sleep(3)
+
+def get_process_stats(pid, end_time_wait, start_time_psutil) -> tuple:
+    try:
+        process = psutil.Process(pid)
+        user_time = process.cpu_times().user
+        system_time = process.cpu_times().system
+        memory_usage = process.memory_info().rss
+        io_counters_raw = process.io_counters()
+        io_counters = {
+            'read_count': io_counters_raw.read_count,
+            'write_count': io_counters_raw.write_count,
+            'read_bytes': io_counters_raw.read_bytes,
+            'write_bytes': io_counters_raw.write_bytes
+        }
+        
+        total_cpu_time = user_time + system_time if user_time and system_time else 0
+        elapsed_time = (end_time_wait - start_time_psutil).total_seconds()
+        cpu_usage_percent = (total_cpu_time / elapsed_time) * 100 if elapsed_time > 0 else 0
+        return user_time, system_time, memory_usage, io_counters, cpu_usage_percent
+    except Exception as e:
+        print(f"Failed to get process stats: {e}")
+        return None, None, None, None
+
+def benchmark_command(command: list[str], warmup_iterations: int, benchmark_iterations: int, transcribe_flag: bool) -> list[Result]:
     results = []
 
     for _ in range(warmup_iterations):
-        subprocess.run(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        print(f"Running warmup command: {' '.join(command)}")
+        subprocess.run("rm probe_log", shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        subprocess.run(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
 
     for _ in range(benchmark_iterations):
+        cleanup()
         start_time_psutil = datetime.datetime.now()
-
-        proc = psutil.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-
-        cpu_times = memory_info = io_counters = None
-
-        if proc.is_running():
-            try:
-                cpu_times = proc.cpu_times()
-                memory_info = proc.memory_info()
-                io_counters = proc.io_counters()
-            except (psutil.NoSuchProcess, psutil.AccessDenied):
-                pass
-
-        stdout, stderr = proc.communicate()
-
-        datetime.datetime.now()
-
-        datetime.datetime.now()
-        returncode = proc.wait()
+        print(f"Starting process with command: {' '.join(command)}")
+        returncode, rusage, stdout, stderr = resource_call(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
         end_time_wait = datetime.datetime.now()
+        
+        pid = os.getpid()  
+        user_time, system_time, memory_usage, io_counters, cpu_usage_percent = get_process_stats(pid, end_time_wait, start_time_psutil)
```

Comment:
I maybe wasn't clear on this earlier. We don't need to use `get_process_stats`, since user time, system time, max memory usage, and IO data is contained in the `rusage` struct.

`ru_utime` is user time, `ru_stime` is system time, the sum is the total CPU time, `ru_maxrss` is the maximum memory usage, `ru_inblock` counts read I/O operations, and `ru_outblock` counts write I/O operations.

Therefore, we don't need `get_process_stats` or `psutil`.

In fact, `resource_call` itself could also measure the wall-time and capture the stdout/stderr streams, since that is the same for every process. It could return a `Result` object itself.